tests/serverpod_test_server/lib/src/endpoints/method_streaming.dart

```diff
@@ -14,15 +14,19 @@ class MethodStreaming extends Endpoint {
     }
   }
 
+  static Completer<StreamController<int>>? neverEndingStreamController;
   Stream<int> neverEndingStreamWithDelay(
     Session session,
     int millisecondsDelay,
-  ) async* {
-    int i = 0;
-    while (true) {
-      await Future.delayed(Duration(milliseconds: millisecondsDelay));
-      yield i++;
-    }
+  ) {
+    var controller = StreamController<int>();
```

Comment:
Question: Is the controller "middle man" needed here to "keep it alive", as opposed to just returning result of Stream.periodic()?

Reply:
This is to enable the test to hook into the stream's lifecycle and assert that it was indeed cancelled in the end.